src/NHibernate/Impl/FilterImpl.cs

```diff
@@ -93,10 +93,19 @@ public IFilter SetParameterList(string name, ICollection values)
 			if (values.Count > 0)
 			{
 				var e = values.GetEnumerator();
-				e.MoveNext();
-				if (!type.ReturnedClass.IsInstanceOfType(e.Current))
+				try
```

Comment:
It looks like is doable without it being a sure breaking-changes: most users are probably already using something implicitly castable to `ICollection<T>`. Changing it does not require any other code change in NHibernate, test project included. Maybe is-it time to purge all our interfaces from old non generic `IEnumerable`, `ICollection`, `IList`. It would be fine to remove them from internal implementations too, but this could be heavier to do.